### DISCUSSION TITLE
README.md Update Handling

### DISCUSSION THREAD

**Discussion by Jotschi** [jreleaser/jreleaser#1305] [Ideas]
I'm currently in the situation in which I would like to include example code snippets and version numbers in my README.md.
Updating version numbers is easy. This can be done via maven resource filtering.
Including source code examples is a bit more involved. The maven site supports this via [Doxia snippet Macro](https://maven.apache.org/guides/mini/guide-snippet-macro.html). But this can't be used for non-site project files.
For Markdown files I only found a no longer maintained maven plugin https://github.com/ryancerf/choss-maven-plugin

Is update handling of MD files already possible with jreleaser or is this out of scope?

**Comment by aalmiray** [comment]
Well JReleaser `1.5.0` now offers a `template eval` command that may be used to create files based on any given template. 

Also, Asciidoc supports including snippets of code with syntax highlighting an other neat features. However, I think that feature may not be enabled when browsing files directly on a GH repository. 

Even more so, if the target format must remain Markdown then Asciidoc is a moot point.

**Reply by aalmiray** [reply]
Moreover this feature might be closer to what a build tool could/should provide rather than a release tool. Though one might be inclined to process doc files during a release ...

**Reply by Jotschi** [reply]
Yes, I think the readme update should be more part of the build process as it is part of the documentation.

I have created a resource filter for the maven-resource-plugin to manage snippet inclusions:

https://github.com/metaloom/snippet-resource-filter
